apps/console/src/features/groups/components/wizard/create-group-wizard-updated.tsx

```diff
@@ -0,0 +1,493 @@
+}
+
+/**
+ * Interface to capture current wizard state
+ */
+interface WizardStateInterface {
+    [ key: string ]: any;
+}
+
+/**
+ * Interface for wizard step.
+ */
+interface WizardStepInterface {
+    content: ReactElement;
+    icon: GenericIconProps | any;
+    name: string;
+    title: string;
+}
+
+/**
+ * Component to handle addition of a new group to the system.
+ *
+ * @param props - props related to the create group wizard Member
+ */
+export const CreateGroupWizardUpdated: FunctionComponent<CreateGroupProps> =
+(props: CreateGroupProps): ReactElement => {
+
+    const {
+        closeWizard,
+        initStep,
+        requiredSteps,
+        [ "data-testid" ]: testId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const [ currentStep, setCurrentWizardStep ] = useState<number>(initStep);
+    const [ partiallyCompletedStep, setPartiallyCompletedStep ] = useState<number>(undefined);
+    const [ wizardState, setWizardState ] = useState<WizardStateInterface>(undefined);
+    const [ wizardSteps, setWizardSteps ] = useState<WizardStepInterface[]>(undefined);
+    const [ selectedUserStore, setSelectedUserStore ] = useState<string>(
+        commonConfig?.primaryUserstoreOnly ? PRIMARY_USERSTORE : CONSUMER_USERSTORE);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    const [ submitGeneralSettings, setSubmitGeneralSettings ] = useTrigger();
+
+    const [ isWizardActionDisabled, setIsWizardActionDisabled ] = useState<boolean>(true);
+
+    const [ alert, setAlert, alertComponent ] = useWizardAlert();
+
+    const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+    useEffect(() => {
+        setWizardSteps(filterSteps([ WizardStepsFormTypes.BASIC_DETAILS ]));
+    }, []);
+
+    /**
+     * Sets the current wizard step to the previous on every `partiallyCompletedStep`
+     * value change , and resets the partially completed step value.
+     */
+    useEffect(() => {
+        if (partiallyCompletedStep === undefined) {
+            return;
+        }
+
+        setCurrentWizardStep(currentStep - 1);
+        setPartiallyCompletedStep(undefined);
+    }, [ partiallyCompletedStep ]);
+
+    /**
+     * Method to handle create group action when create group wizard finish action is triggered.
+     *
+     * @param groupDetails - basic data required to create group.
+     */
+    const addGroup = (groupDetails: any): void => {
+        let groupName: string = "";
+
+        selectedUserStore !== "primary"
+            ? groupName = groupDetails?.basic.basicDetails
+                ? groupDetails?.basic?.basicDetails?.domain + "/" + groupDetails?.basic?.basicDetails?.groupName
+                : groupDetails?.domain + "/" + groupDetails?.groupName
+            : groupName = groupDetails?.basic?.basicDetails
+                ? groupDetails?.basic?.basicDetails?.groupName
+                : groupDetails?.groupName;
```

Comment:
It'd be better if we can simplify nested ternary expressions into if else statements for better readability.

Reply:
Fixed with d08680252dc6b5e81e1864daef86e3756305e27d